src/components/Inputs/SaveData/SaveData.vue

```diff
@@ -49,67 +46,98 @@ export default {
+                voiceMap[key1] = `${keyStrings}-${rId}.wav`;
               }
-              JSONdata[key][key2] = val2;
-            }
-          });
-        } else {
-          JSONdata[key] = val;
-        }
-      });
-      _.map(data.scores, (val, key) => { // todo: check if score object not null?
-        if (!_.isEmpty(val)) {
-          JSONscores[key] = val;
+              // todo: check if sections are present, they are no longer object but lists
+              // else if (_.isObject(value)) {
+              //   // make sure there aren't any Blobs here.
+              //   // if there are, add them to fileUploadData
+              //   _.map(value, (val2, key2) => {
+              //     if (val2 instanceof Blob) {
+              //       // console.log(322, val, key2, val2);
+              //       fileUploadData[`${key2}`] = val2;
+              //     }
+              //     else {
+              //       // refill the object.
+              //       if (!JSONdata[key]) {
+              //         JSONdata[key] = {};
+              //       }
+              //       JSONdata[key][key2] = val2;
+              //     }
+              //   });
+              // }
+              // else {
+              //   JSONdata[key] = val;
+              // }
+            });
+            _.map(voiceMap, (v, ky) => {
+              if (ky in itemObj) {
+                const newObj = itemObj;
+                // console.log(327, itemObj);
+                newObj[ky] = v;
+              }
+            });
+          }
+        });
+        // write out the activity files
+        if (eachActivityList.length) { // if activity is answered then write to file
+          jszip.folder('responses').file(`activity_${key}.jsonld`, JSON.stringify(eachActivityList, null, 4));
+          key += 1;
         }
       });
-      _.map(JSONdata, (val, key) => {
-        jszip.folder('data/responses').file(`activity_${key}.json`, JSON.stringify(val, null, 4));
-      });
-      _.map(JSONscores, (val, key) => {
-        jszip.folder('data/scores').file(`activity_${key}_score.json`, JSON.stringify(val, null, 4));
-      });
-      let f = 0;
-      _.map(fileUploadData, (val) => {
-        jszip.folder('data').file(`voice_item_${f + 1}.wav`, val);
-        f += 1;
-      });
-      // jsonData.score = JSONscores[0];
+
+      // _.map(data.scores, (val, key) => { // todo: check if score object not null?
+      //   if (!_.isEmpty(val)) {
+      //     JSONscores[key] = val;
+      //   }
+      // });
+      // _.map(JSONdata, (val, key) => {
+      //   // console.log(342, (key));
+      // jszip.folder('responses').file(`activity_${key}.json`, JSON.stringify(val, null, 4));
+      // });
+      // _.map(JSONscores, (val, key) => {
+      // jszip.folder('scores').file(`activity_${key}_score.json`, JSON.stringify(val, null, 4));
+      // });
+
+      let registrationURL = `https://sig.mit.edu/vb/token?token=${TOKEN}`;
```

Comment:
i don't understand why this is here. the UI should never be calling that endpoint.